### ISSUE TITLE
Batch umccrise: refdata loading not idempotent

### ISSUE THREAD

**Issue by reisingerf** [umccr/infrastructure#88]
Cloning the refdata repo for pulling refdata is not idempotent and will cause jobs to fail if they get scheduled on instances where the data already exists.

Current:
```bash
git clone https://github.com/umccr/reference_data $REFDATA_DIR
cd $REFDATA_DIR
```

Quick fix proposal:
```bash
if [ -d "$REFDATA_DIR" ]; then
  echo "Refdata dir already exists. Skipping git clone"
else
  echo "Cloning refdata repo"
  git clone https://github.com/umccr/reference_data $REFDATA_DIR
fi
cd $REFDATA_DIR
```

**Comment by reisingerf** [comment]
May need to address possible race condition if `git clone` of the first thread is not finished and the second thread already moves on.
Maybe add an additional `git pull` each thread has to pass through? Or add/check a lock to make sure the `clone` finished?

**Comment by reisingerf** [comment]
And tackle DVC lock...

**Comment by reisingerf** [comment]
Fixed by #90
